pkg/apis/acme/v1/types_issuer.go

```diff
@@ -317,6 +317,14 @@ type ACMEChallengeSolverHTTP01IngressTemplate struct {
 	// will override the in-built values.
 	// +optional
 	ACMEChallengeSolverHTTP01IngressObjectMeta `json:"metadata"`
+
+	// OverrideNginxIngressWhitelistAnnotation allows for overriding the annotation that should be used for whitelist-source-range.
+	// By default, "nginx.ingress.kubernetes.io/whitelist-source-range" is used, but since nginx allows for the annotation namespace
+	// to be customized, the default annotation would in that case be rejected by the nginx validating webhook.
+	// By setting OverrideNginxIngressWhitelistAnnotation to match your nginx annotation namespace this issue will be resolved.
+	// When setting OverrideNginxIngressWhitelistAnnotation, you have to set the full annotation key, i.e:
+	// "ingress.kubernetes.io/whitelist-source-range"
```

Comment:
nit: end with a dot and wrap comments to 80 or 100 chars.

```suggestion
	// OverrideNginxIngressWhitelistAnnotation allows for overriding the
	// annotation that should be used for whitelist-source-range. By default,
	// "nginx.ingress.kubernetes.io/whitelist-source-range" is used, but since
	// nginx allows for the annotation namespace to be customized, the default
	// annotation would in that case be rejected by the nginx validating
	// webhook. By setting OverrideNginxIngressWhitelistAnnotation to match your
	// nginx annotation namespace this issue will be resolved. When setting
	// OverrideNginxIngressWhitelistAnnotation, you have to set the full
	// annotation key, i.e: "ingress.kubernetes.io/whitelist-source-range".
```